src/main/scala/mesosphere/marathon/MarathonModule.scala

```diff
@@ -348,7 +351,7 @@ class MarathonModule(conf: MarathonConf, http: HttpConf)
   def provideAppRepository(
     @Named(ModuleNames.STORE_APP) store: EntityStore[AppDefinition],
     metrics: Metrics): AppRepository = {
-    new AppRepository(store, maxVersions = conf.zooKeeperMaxVersions.get, metrics)
+    new AppEntityRepository(store, maxVersions = conf.zooKeeperMaxVersions.get, metrics)
```

Comment:
Ideally we get rid of all store/repository objects here and make them available via the storage module (no guice)